tests/unit/components/listing/FfaListingComponent.spec.ts

```diff
@@ -0,0 +1,123 @@
+import appStore from '../../../../src/store'
+import FfaListing, { FfaListingStatus} from '../../../../src/models/FfaListing'
+
+const localVue = createLocalVue()
+const ffaListingClass = '.ffa-listing'
+const userAddress = '0xwall3t'
+
+describe('FfaListingComponent.vue', () => {
+
+  beforeAll(() => {
+    localVue.use(VueRouter)
+  })
+
+  describe('handleDisplay()', () => {
+    it('correctly renders user candidate listings when given address and candidate props', async () => {
+      const wrapper = shallowMount(FfaListingComponent, {
+        attachToDocument: true,
+        store: appStore,
+        propsData: {
+          status: FfaListingStatus.candidate,
+          userAddress,
+        },
+      })
+      // @ts-ignore
+      await wrapper.vm.handleDisplay()
+
+      const userCandidatesOnly = wrapper.vm.$data.displayedListings.every((candidate: FfaListing) => {
+        return candidate.status === FfaListingStatus.candidate && candidate.owner === userAddress
+      })
+
+      expect(wrapper.findAll(`${ffaListingClass}`).length).toBe(5)
+      expect(userCandidatesOnly).toBeTruthy()
+    })
+
+    it('correctly renders user listed listings when given address and listed props', async () => {
+      const wrapper = shallowMount(FfaListingComponent, {
+        attachToDocument: true,
+        store: appStore,
+        propsData: {
+          status: FfaListingStatus.listed,
+          userAddress,
+        },
+      })
+      // @ts-ignore
+      await wrapper.vm.handleDisplay()
+
+      const userListedOnly = wrapper.vm.$data.displayedListings.every((candidate: FfaListing) => {
+        return candidate.status === FfaListingStatus.listed && candidate.owner === userAddress
+      })
+
+      expect(wrapper.findAll(`${ffaListingClass}`).length).toBe(4)
+      expect(userListedOnly).toBeTruthy()
+    })
+
+    it('correctly renders all user listings when given address and listed props', async () => {
+      const wrapper = shallowMount(FfaListingComponent, {
+        attachToDocument: true,
+        store: appStore,
+        propsData: {
+          status: '',
+          userAddress,
+        },
+      })
+      // @ts-ignore
+      await wrapper.vm.handleDisplay()
+
+      const userListingsOnly = wrapper.vm.$data.displayedListings.every((candidate: FfaListing) => {
+        return candidate.owner === userAddress
+      })
+
+      expect(wrapper.findAll(`${ffaListingClass}`).length).toBe(9)
+      expect(userListingsOnly).toBeTruthy()
+    })
+
+    it('correctly renders all candidate listings when given candidate props', async () => {
+      const wrapper = shallowMount(FfaListingComponent, {
+        attachToDocument: true,
+        store: appStore,
+        propsData: { status: FfaListingStatus.candidate },
+      })
+      // @ts-ignore
+      await wrapper.vm.handleDisplay()
+
+      const candidatesOnly = wrapper.vm.$data.displayedListings.every((candidate: FfaListing) => {
```

Comment:
while this is clever, i don't think it's testing the right thing.  since we're mounting the component we should be testing the wrapper output, which means testing the html.  if you add `data-` attributes to the item `<div>`s you should be able to select those out.